syft/rekor/query.go

```diff
@@ -0,0 +1,182 @@
+package rekor
+
+import (
+	"context"
+	"crypto"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/source"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/client/index"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spdx/tools-golang/spdx"
+)
+
+// getSbom attempts to retrieve the SBOM from the URI.
+//
+// Precondition: att is not nil and client is not nil
+func getSbom(att *InTotoAttestation, client *http.Client) (*[]byte, error) {
```

Comment:
I use the bytes later on in different functions (to hash and to load into a spdx struct). Would it be better return an io.Reader and then copy it so it can be used twice?